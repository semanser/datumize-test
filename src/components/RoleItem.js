import React from 'react'
import AvatarImage from './AvatarImage'

const RoleItem = props => (
  <div className="roles-item" onClick={props.onSelect}>
    <AvatarImage type="role" name={props.role.name}/>
    <div className="text">{props.role.name}</div>
  </div>
)

export default RoleItem

