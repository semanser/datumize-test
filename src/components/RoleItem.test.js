import React from 'react'
import ReactDOM from 'react-dom'
import ShallowRenderer from 'react-test-renderer/shallow'
import RoleItem from './RoleItem'

const renderer = new ShallowRenderer()

describe('RoleItem Component', () => {
  it('Renders component without crashing', () => {
    const div = document.createElement('div')
    const props = {
      role: {
        id: 1,
        name: 'Admin',
      }
    }

    const tree = renderer.render(<RoleItem {...props}/>)
    expect(tree).toMatchSnapshot()
  })
})

