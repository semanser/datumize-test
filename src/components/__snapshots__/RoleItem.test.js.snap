// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleItem Component Renders component without crashing 1`] = `
<div
  className="roles-item"
  onClick={undefined}
>
  <AvatarImage
    name="Admin"
    type="role"
  />
  <div
    className="text"
  >
    Admin
  </div>
</div>
`;
